File: Java/Desafio4 - GFT/src/br/com/gft/model/Loja.java
package br.com.gft.model;

import java.util.List;

public class Loja {

	private String nome, cnpj;
	private List<Livro> livros;
	private List<VideoGame> videoGame;
	
	public String getNome() {
		return nome;
	}
	public void setNome(String nome) {
		this.nome = nome;
	}
	public String getCnpj() {
		return cnpj;
	}
	public void setCnpj(String cnpj) {
		this.cnpj = cnpj;
	}
	public List<Livro> getLivros() {
		return livros;
	}
	public void setLivros(List<Livro> livros) {
		this.livros = livros;
	}
	public List<VideoGame> getVideoGame() {
		return videoGame;
	}
	public void setVideoGame(List<VideoGame> videoGame) {
		this.videoGame = videoGame;
	}
	
	public Loja() {
	}
	
	public Loja(String nome, String cnpj, List<Livro> livros, List<VideoGame> videoGame) {
		super();
		this.nome = nome;
		this.cnpj = cnpj;
		this.livros = livros;
		this.videoGame = videoGame;
	}
	
	public void listaLivros() {
		System.out.println("\n====================================================================================================");
		System.out.println("Lista de Livros da loja " + getNome());
		if (livros.isEmpty()) {
			System.out.println("A loja não tem livros no seu estoque");
		} else {
			//estrutura padrão do For
			for (int i = 0; i < livros.size(); i++) {
				System.out.printf("Título: %s, Preço: R$ %.2f, Quantidade: %d, Autor: %s, Tema: %s, Páginas: %d \n", livros.get(i).getNome(),
						livros.get(i).getPreco(), livros.get(i).getQtd(), livros.get(i).getAutor(), livros.get(i).getTema(), livros.get(i).getQtdPag());
			}
		}
		System.out.println("====================================================================================================\n");
	}
	
	public void listaVideoGames() {
		System.out.println("====================================================================================================");
		System.out.println("Lista de Videogames da loja " + getNome());
		if (livros.isEmpty()) {
			System.out.println("A loja não tem video-games no seu estoque");
		} else {	
			//Utilizando forEach
			for (VideoGame dados : videoGame) {
				String usado;
				if (dados.getUsado() == true) {
					usado = "sim";
				} else {
					usado = "não";
				}
				System.out.printf("Videogame: %s, Preço: R$ %.2f, Quantidade: %d, Marca: %s, Modelo: %s, Usado: %s \n", dados.getNome(), 
						dados.getPreco(), dados.getQtd(),dados.getMarca(), dados.getModelo(), usado);
			}
		}
		System.out.println("====================================================================================================\n");
	}
	
	public double calculaPatrimonio(){
		double soma = 0;
		for (VideoGame game : videoGame) {
			soma += game.getPreco()*game.getQtd();
		}
		for (Livro livro : livros) {
			soma += livro.getPreco()*livro.getQtd();
		}
		return soma;
	}

}

File: C-Sharp/Desafio04-C-Sharp/Desafio04-C-Sharp/com.gft.model/Loja.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Desafio04_C_Sharp.com.gft.model
{
    class Loja
    {
		public string Nome { get; set; }
		public string Cnpj { get; set; }
		internal List<Livro> Livros { get; set; }
		internal List<VideoGame> VideoGame { get; set; }

		public Loja()
        {
        }

		public Loja(string nome, string cnpj, List<Livro> livros, List<VideoGame> videoGame)
		{
			Nome = nome;
			Cnpj = cnpj;
			Livros = livros;
			VideoGame = videoGame;
		}

		public void listaLivros()
		{
			Console.WriteLine("====================================================================================================");
			Console.WriteLine("Lista de Livros da loja " + Nome);
			if (Livros.Count == 0)
			{
				Console.WriteLine("A loja não tem livros no seu estoque");
			}
			else
			{
				foreach (Livro dados in Livros)
				{
					Console.WriteLine("Título: {0}, Preço: R$ {1:0.00}, Quantidade: {2}, Autor: {3}, Tema: {4}, Páginas: {5}", dados.Nome,
							dados.Preco, dados.Qtd, dados.Autor, dados.Tema, dados.QtdPag);
				}
			}
			Console.WriteLine("====================================================================================================");
		}

		public void listaVideoGames()
		{
			Console.WriteLine("====================================================================================================");
			Console.WriteLine("Lista de Videogames da loja " + Nome);
			if (VideoGame.Count == 0)
			{
				Console.WriteLine("A loja não tem video-games no seu estoque");
			}
			else
			{
				//Utilizando forEach
				foreach (VideoGame dados in VideoGame)
				{
					String usado;
					if (dados.IsUsado == true)
					{
						usado = "sim";
					}
					else
					{
						usado = "não";
					}
					Console.WriteLine("Videogame: {0}, Preço: R$ {1:0.00}, Quantidade: {2}, Marca: {3}, Modelo: {4}, Usado: {5}", dados.Nome,
							dados.Preco, dados.Qtd, dados.Marca, dados.Modelo, usado);
				}
			}
			Console.WriteLine("====================================================================================================");
		}

		public double calculaPatrimonio()
		{
			double soma = 0;
			foreach (VideoGame game in VideoGame)
			{
				soma += game.Preco * game.Qtd;
			}
			foreach (Livro livro in Livros)
			{
				soma += livro.Preco * livro.Qtd;
			}
			return soma;
		}
	}
}
